pkg/appgw/requestroutingrules.go

```diff
@@ -130,10 +130,16 @@ func (c *appGwConfigBuilder) noRulesIngress(cbCtx *ConfigBuilderContext, ingress
 	if err != nil {
 		glog.Error("Error fetching Backends and Settings: ", err)
 	}
+
+	noRulesIngressHTTPSettings := c.generateHTTPSettings(backendID, serviceBackendPairMap[backendID].BackendPort, cbCtx)
+
+	// Thre are no Rules, create urlPathMap with no pathRules
 	if serviceBackendPair, exists := serviceBackendPairMap[backendID]; exists {
 		poolName := generateAddressPoolName(backendID.serviceFullName(), backendID.Backend.ServicePort.String(), serviceBackendPair.BackendPort)
 		defaultAddressPoolID := c.appGwIdentifier.AddressPoolID(poolName)
-		defaultHTTPSettingsID := c.appGwIdentifier.HTTPSettingsID(DefaultBackendHTTPSettingsName)
+		//create default HttpSettings from ingress default backend
+		defaultHTTPSettingsID := c.appGwIdentifier.HTTPSettingsID(*noRulesIngressHTTPSettings.Name)
+
 		listenerID := defaultFrontendListenerIdentifier()
```

Comment:
I think a similar change would be needed for frontend listener also. That is if the front end listener is having hostname-extension. Then default listener might not work. We can address this in a separate PR as well.